.github/workflows/release.yml

```diff
@@ -0,0 +1,94 @@
+name: Release
+
+on: [workflow_dispatch]
+
+jobs:
+  test:
+    name: "Test on Python ${{ matrix.python-version }} (${{ matrix.os }})"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
+    steps:
+      - uses: "actions/checkout@v4"
+        with:
+          fetch-depth: 0
+
+      - uses: "actions/setup-python@v5"
+        with:
+            python-version: "${{ matrix.python-version }}"
+
+      - name: "Installs for ${{ matrix.python-version }}"
+        run: |
+          pip install --upgrade pip
+          pip install nox
+
+      - name: "Run nox for Python ${{ matrix.python-version }}"
+        run: "nox -s test-${{ matrix.python-version }}"
+
+  build:
+    name: Build wheel and sdist
+    needs: test
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+      
+      - name: Install build dependency
```

Comment:
change to flit instead of build

Reply:
So what build does is it reads the `[build-system]` information in `pyproject.toml` and then uses flit to build the project. It means if you decide to change the build system for whatever reason the action will still work.

Reply:
good points